runtime/parachains/src/paras_inherent.rs

```diff
@@ -80,24 +81,13 @@ impl WeightInfo for TestWeightInfo {
 }
 
 fn paras_inherent_total_weight<T: Config>(
-	n_backed_candidates: u32,
-	n_bitfields: u32,
-	n_disputes: u32,
+	backed_candidates: &[BackedCandidate<<T as frame_system::Config>::Hash>],
+	bitfields: &[UncheckedSignedAvailabilityBitfield],
+	disputes: &[DisputeStatementSet],
 ) -> Weight {
-	<T as Config>::WeightInfo::enter_variable_disputes(MAX_ACTIVE_VALIDATORS)
-		.saturating_mul(n_disputes as Weight)
-		.saturating_add(
-			<T as Config>::WeightInfo::enter_backed_candidates_variable(MAX_ACTIVE_VALIDATORS)
-				.saturating_mul(n_backed_candidates as Weight),
-		)
-		.saturating_add(
-			<T as Config>::WeightInfo::enter_bitfields().saturating_mul(n_bitfields as Weight),
-		)
-}
-
-/// Extract the weight that is added _per_ bitfield.
-fn bitfield_weight<T: Config>() -> Weight {
-	<T as Config>::WeightInfo::enter_bitfields()
+	backed_candidates_weight::<T>(backed_candidates) +
+	signed_bitfields_weight::<T>(bitfields.len()) +
+	dispute_statements_weight::<T>(disputes)
```

Comment:
Always use `saturating_add` for this stuff like this so we don't panic

Reply:
Nice catch